test/functional/buildAndPackage/src/net/adoptopenjdk/test/CudaEnabledTest.java

```diff
@@ -4,96 +4,125 @@
-                prtLibDirectory += "/bin/default";
+
+        Set<String> j9prtFiles = Files.walk(Paths.get(testJdkHome))
+                                      .map(Path::toString)
+                                      .filter(name -> j9prtPattern.matcher(name).matches())
+                                      .collect(Collectors.toSet());
+
+        Assert.assertFalse(j9prtFiles.isEmpty(), "Can't find a j9prt file anywhere in " + testJdkHome);
+
+        for (String prtFile : j9prtFiles) {
+            Assert.assertTrue((new File(prtFile)).exists(), "Found the prt file, but it doesn't exist. Tautology bug.");
+            Assert.assertTrue((new File(prtFile)).canRead(), "Found the prt file, but it can't be read. Likely a permissions bug.");
+            logger.info("j9prt file identified: " + prtFile);
+        }
+
+        //Stage 2: Iterate through the j9prt files to find "cudart".
+        //If we find it in every j9prt file, then cuda functionality is enabled on every j9 vm in this build.
+        int successes = 0;
+        for (String prtFile : j9prtFiles) {
+            try {
+                BufferedReader prtFileReader = new BufferedReader(new FileReader(prtFile));
+                String oneLine = "";
+                boolean foundCudart = false;
+                while ((oneLine = prtFileReader.readLine()) != null) {
+                    if (oneLine.contains("cudart")) {
+                        logger.info("Cuda-enabled indicator string \'cudart\' was found within this j9prt file: " + prtFile);
+                        foundCudart = true;
+                        break;
+                    }
+                }
+                prtFileReader.close();
+                if (foundCudart) {
+                    successes++;
+                    continue;
+                }
+            } catch (FileNotFoundException e) {
+                Assert.fail("A file that exists could not be found. This should never happen.");
+            } catch (Exception e) {
+                throw new Error(e);
             }
+
+            logger.info("Cuda-enabled indicator string \'cudart\' was not found within this j9prt file: " + prtFile);
         }
 
-        File prtDirObject = new File(prtLibDirectory);
-        Assert.assertTrue(prtDirObject.exists(), "Can't find the predicted location of the j9prt lib file. Expected location: " + prtLibDirectory);
-        
-        String[] prtLibDirectoryFiles = prtDirObject.list();
-        String prtFile = null;
-        for(int x = 0 ; x < prtLibDirectoryFiles.length ; x++) {
-        	if(prtLibDirectoryFiles[x].contains("j9prt")) {
-        	    prtFile = prtLibDirectory + "/" + prtLibDirectoryFiles[x];
-        	    break;
-        	}
+        logger.info("j9prt file scanning complete. Assessing results.");
+
+        if ((successes > 0) && (successes < j9prtFiles.size())) {
+            Assert.fail("Multiple j9prt files were found, some with cuda enabled and some without.");
         }
```

Comment:
I think this should be removed in favour of the assertions below.

Reply:
Done.